src/anim/evaluator/anim-clip.js

```diff
@@ -25,6 +25,12 @@ class AnimClip {
         this._loop = loop;              // whether to loop
         this._blendWeight = 1.0;        // blend weight 0..1
         this._blendOrder = 0.0;         // blend order relative to other clips
+        this._eventHandler = eventHandler;
+        this._eventCursor = 0;
+        // move the event cursor to an event that should fire after the starting time
+        while (this._track.events[this._eventCursor]?.time < this.time) {
```

Comment:
not too keen on using `?.` here. For out of array access this returns _undefined_ which you then compare to time. It works and returns false, but the strongly typed part of my mind does not like it :)